internal/channel/plugin.go

```diff
@@ -1,19 +1,86 @@
 package channel
 
 import (
+	"bufio"
 	"encoding/json"
-	"errors"
 	"fmt"
-	"github.com/icinga/icinga-notifications/internal/contracts"
-	"github.com/icinga/icinga-notifications/internal/event"
-	"github.com/icinga/icinga-notifications/internal/recipient"
+	"github.com/icinga/icinga-notifications/internal/daemonConfig"
 	"github.com/icinga/icinga-notifications/pkg/plugin"
 	"github.com/icinga/icinga-notifications/pkg/rpc"
-	"strings"
+	"github.com/icinga/icingadb/pkg/icingadb"
+	"github.com/icinga/icingadb/pkg/logging"
+	"go.uber.org/zap"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"time"
 )
 
-func (c *Channel) GetInfo() (*plugin.Info, error) {
-	result, err := c.rpcCall(plugin.MethodGetInfo, nil)
+type Plugin struct {
+	cmd    *exec.Cmd
+	rpc    *rpc.RPC
+	mu     sync.Mutex
+	logger *zap.SugaredLogger
+}
+
+// NewPlugin starts and returns a new plugin instance. If the start of the plugin fails, an error is returned
+func NewPlugin(pluginName string, logger *zap.SugaredLogger) (*Plugin, error) {
+	p := &Plugin{logger: logger}
+	err := p.Start(pluginName)
```

Comment:
There is no reason to call `Start()` anywhere else, is it? Then it should not be exported, or maybe even just merged in this function as this doesn't do much more as calling `Start()` anyways.